evalml/tests/automl_tests/test_automl.py

```diff
@@ -5047,6 +5054,60 @@ def test_default_algorithm_uses_n_jobs(X_y_binary, AutoMLTestEnv):
     assert n_checked and n_feature_selector_checked
 
 
+@pytest.mark.parametrize("input_type", ["pd", "ww"])
+@pytest.mark.parametrize("automl_algorithm", ["default", "iterative"])
+@pytest.mark.parametrize("problem_type", ProblemTypes.all_problem_types)
+def test_exclude_featurizers(
```

Comment:
could you repurpose this test to run search and then check the pipelines generated by search with `get_pipeline()`? I think this current test would be better suited in both `test_default_algorithm` and `test_iterative_algorithm` separately where we check pipeline generation with each algorithm.  I general I try to separate testing top level search behavior in this file and then algorithm specific pipeline generation logic in each algorithm's test file. You can use our `AutoMLTestEnv` fixture to skip computation here. Here's a good [example](https://github.com/alteryx/evalml/blob/main/evalml/tests/automl_tests/test_automl.py#L235-L237).

Reply:
@jeremyliweishih Just to be clear on what you are proposing.

You think I should split the existing test into two, and move the first case for default algorithm into `test_default_algorithm.py` and the second iterative case into `test_iterative_algorithm.py`. Then, create a new test in `test_automl.py` that uses the `AutoMLTestEnv` fixture like the example you linked? Is that correct?

Reply:
@thehomebrewnerd yes! how does that sound?

Reply:
Works for me! I'll work on that update this afternoon.

Reply:
@jeremyliweishih I'm having a little trouble with using `AutoMLTestEnv`, so might need some pointers next week on how to resolve. This commit contains my attempt at updating the test in `test_automl.py`: [c9c3377](https://github.com/alteryx/evalml/pull/3631/commits/c9c3377dfe15103c1e722cbb8f6dfabb7d4a674d)

Is that what you were envisioning?

Reply:
yup! it looks basically there. Please ping me next week!

Reply:
Thanks - will do. This is the error I am getting, but haven't really dug too deep to understand what is happening.

```
evalml/automl/automl_search.py:1064: in search
    pipeline_id = self._post_evaluation_callback(
evalml/automl/automl_search.py:1388: in _post_evaluation_callback
    self.automl_algorithm.add_result(
evalml/automl/automl_algorithm/default_algorithm.py:485: in add_result
    self._parse_selected_categorical_features(pipeline)
evalml/automl/automl_algorithm/default_algorithm.py:431: in _parse_selected_categorical_features
    self._get_feature_provenance_and_remove_engineered_features(
evalml/automl/automl_algorithm/default_algorithm.py:411: in _get_feature_provenance_and_remove_engineered_features
    component = pipeline.get_component(component_name)
evalml/pipelines/pipeline_base.py:224: in get_component
    return self.component_graph.get_component(name)

FAILED evalml/tests/automl_tests/test_automl.py::test_exclude_featurizers[binary-default-pd] - ValueError: Component URL Featurizer is not in the graph
```